### PR TITLE
Support tagged CodeInstance

### PR DESCRIPTION
**Pull request by vchuravy**
Support JuliaLang/julia#52233

### CODE REVIEW

File: src/interpreter.jl [hunk 1]
```diff
@@ -65,6 +65,13 @@ end
 #=CC.=#get_inference_world(interp::CthulhuInterpreter) = get_inference_world(interp.native)
 CC.get_inference_cache(interp::CthulhuInterpreter) = get_inference_cache(interp.native)
 
+struct CthulhuCacheToken
+    native::Any
+end
+@static if isdefined(CC, :cache_owner)
+    CC.cache_owner(interp::CthulhuInterpreter) = CthulhuCacheToken(CC.cache_owner(interp.native))
```
Comment (vchuravy) [inline src/interpreter.jl:72]: We could in theory also use the cache of the AbsInt we are wrapping, but that seems dangerous.

**Comment by vchuravy** [comment]
@Keno @aviatesk this fails in an interesting way.

```
#5  0x00007f9b909a4d26 in __assert_fail (assertion=0x7f9b8fcdf250 "jl_is_string(data)", 
    file=0x7f9b8fcdef00 "/home/vchuravy/src/julia2/src/ircode.c", line=993, 
    function=0x7f9b8fce0040 <__PRETTY_FUNCTION__.10> "ijl_ir_flag_inferred") at assert.c:101
#6  0x00007f9b8fb29b88 in ijl_ir_flag_inferred (data=0x7f9b84b47a80) at /home/vchuravy/src/julia2/src/ircode.c:993
#7  0x00007f9b8fa48686 in _jl_rettype_inferred (owner=0x7f9b71fd3360, mi=0x7f9b7cff48b0 <jl_system_image_data+26621296>, min_world=26726, 
    max_world=26726) at /home/vchuravy/src/julia2/src/gf.c:448
#8  0x00007f9b8fa486f6 in jl_rettype_inferred (owner=0x7f9b71fd3360, mi=0x7f9b7cff48b0 <jl_system_image_data+26621296>, min_world=26726, 
    max_world=26726) at /home/vchuravy/src/julia2/src/gf.c:458
--Type <RET> for more, q to quit, c to continue without paging--
#9  0x00007f9b73f87d68 in get () at compiler/cicache.jl:57
#10 julia_typeinf_edge_3361 (interp=..., method=..., atype=0x7f9b7cff79f0 <jl_system_image_data+26633904>, sparams=0x1, caller=...)
    at compiler/typeinfer.jl:824
```

and

```
(rr) p jl_(jl_typeof(data))
Cthulhu.OptimizedSource
```

in https://github.com/JuliaLang/julia/blob/e50778520b4ec7db350607188dc18bc6c0719b94/src/ircode.c#L993

**Comment by vchuravy** [comment]
@aviatesk mentioned over slack that we should move to using `analysis_results`

**Comment by aviatesk** [comment]
Looks like this should now be closed.

**Pull request closed without merging**